core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1854,13 +1900,10 @@ object SnappySession extends Logging {
         case a: Alias =>
           Alias(a.child, a.name)(exprId = ExprId(0))
         case l@ListQuery(query, _) =>
-          val xx = l.copy(query =
-              query.transform(transformExprID),
+          l.copy(query = query.transformAllExpressions(normalizeExprIds),
```

Comment:
Yes we do. In fact that was leading to a lot of problems. List elements can themselves be expressions (ParamLiterals)